aptos-move/aptos-e2e-comparison-testing/src/data_stateview.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_language_e2e_tests::data_store::FakeDataStore;
+use aptos_types::{
+    state_store::{
+        state_key::StateKey, state_storage_usage::StateStorageUsage, state_value::StateValue,
+        Result as StateViewResult, TStateView,
+    },
+    transaction::Version,
+};
+use aptos_validator_interface::{AptosValidatorInterface, DebuggerStateView};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+pub struct DataStateView {
+    debugger_view: DebuggerStateView,
+    code_data: Option<FakeDataStore>,
```

Comment:
Why use `FakeDataStore`? What does this code data do? I see we pass it into `new` but how it gets populated?

Reply:
Sorry for the confusion. DataStateView is currently used in two places: 1) in the dump mode, I used data_read_state_keys to store the data state value, which is then dumped to the file; 2) in the online mode, I use code_data to store the bytecode compiled by V1 and V2, whenever a txn is obtained, I will use debugger_view + code_data to execute the txn. It is used in the `get_state_value` function as long as it is not none